snforge_std/src/cheatcodes/events.cairo

```diff
@@ -124,38 +128,35 @@ fn remove_event(ref self: EventSpy, index: usize) {
         }
 
         if k != index {
-            emitted_events_deleted_event.append(copy_event(emitted_events.at(k)));
+            let (from, event) = emitted_events.at(k);
+            emitted_events_deleted_event.append((*from, event.clone()));
         }
         k += 1;
     };
     self.events = emitted_events_deleted_event;
 }
 
-fn copy_event(event: @Event) -> Event {
-    let from = *event.from;
-    let name = *event.name;
-
-    let mut keys = array![];
-    let mut i = 0;
-    loop {
-        if i >= event.keys.len() {
-            break;
-        }
+impl EventTraitImpl of starknet::Event<Event> {
```

Comment:
```suggestion
impl EventTraitImpl of starknet::Event<T> {
```
Couldn't we use the implementation that is generated when using `#[derive(starknet::Event)]` instead of providing our generic implementation? Doesn't it lead to conflicting implementations of the trait?